sdks/java/harness/src/test/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClientTest.java

```diff
@@ -146,20 +147,25 @@ public void testLogging() throws Exception {
               apiServiceDescriptor,
               (Endpoints.ApiServiceDescriptor descriptor) -> channel);
 
+      // Keep a strong reference to the loggers in this block. Otherwise the call to client.close()
+      // removes the only reference and the logger may get GC'd (BEAM-4136).
+      Logger rootLogger = LogManager.getLogManager().getLogger("");
+      Logger configuredLogger = LogManager.getLogManager().getLogger("ConfiguredLogger");
+
       // Ensure that log levels were correctly set.
-      assertEquals(Level.OFF, LogManager.getLogManager().getLogger("").getLevel());
-      assertEquals(Level.FINE, LogManager.getLogManager().getLogger("ConfiguredLogger").getLevel());
+      assertEquals(Level.OFF, rootLogger.getLevel());
+      assertEquals(Level.FINE, configuredLogger.getLevel());
 
       // Should be filtered because the default log level override is OFF
-      LogManager.getLogManager().getLogger("").log(FILTERED_RECORD);
+      rootLogger.log(FILTERED_RECORD);
       // Should not be filtered because the default log level override for ConfiguredLogger is DEBUG
-      LogManager.getLogManager().getLogger("ConfiguredLogger").log(TEST_RECORD);
-      LogManager.getLogManager().getLogger("ConfiguredLogger").log(TEST_RECORD_WITH_EXCEPTION);
+      configuredLogger.log(TEST_RECORD);
+      configuredLogger.log(TEST_RECORD_WITH_EXCEPTION);
       client.close();
 
       // Verify that after close, log levels are reset.
-      assertEquals(Level.INFO, LogManager.getLogManager().getLogger("").getLevel());
-      assertNull(LogManager.getLogManager().getLogger("ConfiguredLogger").getLevel());
+      assertEquals(Level.INFO, rootLogger.getLevel());
+      assertNull(configuredLogger.getLevel());
 
       assertTrue(clientClosedStream.get());
       assertTrue(channel.isShutdown());
```

Comment:
Out of time for today, but I just noticed this can happen in other tests as well. For example in [this run](https://github.com/apache/beam/pull/13128/checks?check_run_id=1398707923) we see the same flake in `testWhenServerHangsUpEarlyThatClientIsAbleCleanup`

Leaving a note here so I remember to fix it later

Reply:
ok I fixed the other tests as well, PTAL